library/spdm_responder_lib/libspdm_rsp_encap_response.c

```diff
@@ -198,30 +196,47 @@ void libspdm_init_mut_auth_encap_state(libspdm_context_t *spdm_context, uint8_t
 
     /* Possible Sequence:
      * 2. Session Mutual Auth: (spdm_context->last_spdm_request_session_id_valid)
-     *    2.1 GET_DIGEST/GET_CERTIFICATE (spdm_context->encap_context.req_slot_id != 0xFF)
-     *    2.2 GET_DIGEST (spdm_context->encap_context.req_slot_id == 0xFF)
-     *    2.3 N/A (SPDM_GET_CAPABILITIES_REQUEST_FLAGS_PUB_KEY_ID_CAP)*/
+     *    2.1 GET_DIGEST/GET_CERTIFICATE (MUT_AUTH_REQUESTED_WITH_ENCAP_REQUEST, encap_context.req_slot_id must not be 0xFF)
+     *    2.2 GET_DIGEST (MUT_AUTH_REQUESTED_WITH_GET_DIGESTS, encap_context.req_slot_id must not be 0xFF)
```

Comment:
Hi @jyao1 , I don't see how this 2.2 sequence comes here.
According to SPDM specification v1.2, section 10.22.2 Optimized encapsulated request flow, paragraph 610, the `Optimized encapsulated request flow` (case `SPDM_KEY_EXCHANGE_RESPONSE_MUT_AUTH_REQUESTED_WITH_GET_DIGESTS`) just makes the Requester be able to skip the initial `GET_ENCAPSULATED_REQUEST` to get encapsulated `GET_DIGESTS` request from the Responder. And the encapsulated `GET_CERTIFICATE` flow afterwards may still needed, as the Figure 22 — Optimized session-based mutual authentication example shows.

Though the SDPM specification dosen't have clear description about that encapsulated `GET_CERTIFICATE` flow can be skipped in mutual authentication process if encapsulated `GET_DIGESTS` result is good, just as same as the one-way authentication flow does, I'm just guessing the idea here is to handle such scenario. However I think we would never know it at here at this moment but after the encapsulated `DIGESTS` response is processed by the Responder.

Reply:
you are right.  my mistake. Flow is updated.